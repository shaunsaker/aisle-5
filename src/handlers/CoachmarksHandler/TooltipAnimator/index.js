import React from 'react';
import PropTypes from 'prop-types';
import { ViewPropTypes } from 'react-native';
import Animator from 'react-native-simple-animators';

import styleConstants from '../../../styleConstants';

const propTypes = {
  children: PropTypes.node,
  style: ViewPropTypes.style,
};

const defaultProps = {};

const TooltipAnimator = ({ children, style }) => {
  return (
    <Animator
      type="opacity"
      initialValue={0}
      finalValue={1}
      delay={500}
      shouldAnimateIn
      easing={styleConstants.easing}
      style={style}
    >
      <Animator
        type="scale"
        initialValue={0}
        finalValue={1}
        shouldAnimateIn
        delay={500}
        easing={styleConstants.easing}
      >
        <Animator
          type="scale"
          initialValue={1}
          finalValue={1.05}
          shouldAnimateIn
          shouldRepeat
          duration={600}
          easing={styleConstants.easing}
        >
          {children}
        </Animator>
      </Animator>
    </Animator>
  );
};

TooltipAnimator.propTypes = propTypes;
TooltipAnimator.defaultProps = defaultProps;

export default TooltipAnimator;
