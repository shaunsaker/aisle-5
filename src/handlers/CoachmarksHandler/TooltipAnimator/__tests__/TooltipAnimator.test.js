import React from 'react';
import renderer from 'react-test-renderer';

import TooltipAnimator from '..';

jest.useFakeTimers();

describe('TooltipAnimator', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<TooltipAnimator />);

      expect(component).toMatchSnapshot();
    });
  });
});
