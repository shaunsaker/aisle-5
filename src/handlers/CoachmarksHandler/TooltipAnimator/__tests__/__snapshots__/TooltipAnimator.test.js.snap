// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipAnimator renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    />
  </View>
</View>
`;
